### PR TITLE
Inferno wrapper for jquery

### PR DESCRIPTION
**Pull request by MikeVitik**

### CODE REVIEW

File: packages/jquery/gulpfile.mjs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import gulp from 'gulp';
+import fs from 'node:fs';
+import { generateInfernoFromReactComponents, ReactSrc } from './gulp/inferno-from-react.mjs';
+import './gulp/js-bundles.mjs';
+
+gulp.task('build-dev', gulp.series(
+   function generateInfernoComponents() {
+      return generateInfernoFromReactComponents('./src/generated');
+   },
+   function patchGeneratedSourcies(done) {
```
Comment (const314) [inline packages/jquery/gulpfile.mjs]: Typo: rename patchGeneratedSourcies to patchGeneratedSources

**Comment by williamvinogradov** [comment]
The npm command ```npm run update:tsconfig``` fails.
Also, if I understood right, you touch by hand the auto-generated ts config file  
```packages/devextreme-core/tsconfig.package.json```

So, here are two options:
- If you need special ts configs, let's remove jQuery from the ts config generation like it was done in [this PR](https://github.com/DevExpress/DevExtreme/pull/23407/files#diff-9fd70f89265b560c377505977dd053cc8de9b67ba5b2d17ead471a5035057b72)
- If the basic ts config for jquery is okay, let's add package.json to the ```playgrounds/jQuery``` and generate ```packages/devextreme-core/tsconfig.package.json``` by npm command.

**It'll be cool to do it separately from this PR.**

File: packages/jquery/gulp/js-bundles.mjs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    stats: 'verbose',
+    module: {
+        rules: [
+            {
+                test: /\.ts$/,
+                use: [
+                    {
+                        loader: 'ts-loader',
+                        options: {
+                            configFile: fileURLToPath(new URL('./tsconfig.json', import.meta.url)) // path.resolve(dirname, './tsconfig.json'),
```
Comment (williamvinogradov) [inline packages/jquery/gulp/js-bundles.mjs]: Minor comment.

Let's remove this comment at the end of the line :)

File: packages/jquery/gulp/tsconfig.json [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+        "sourceMap": true,
+        "declaration": false,
+        "downlevelIteration": true,
+        "emitDecoratorMetadata": true,
+        "experimentalDecorators": true,
+        "allowSyntheticDefaultImports": true,
+        "moduleResolution": "node",
+        "jsx": "preserve",
+        "importHelpers": true,
+        "target": "ES2019",
```
Comment (williamvinogradov) [inline packages/jquery/gulp/tsconfig.json]: Question.

Why do we have ```"target": "es2019"```?

File: packages/jquery/package.json [hunk 4]
```diff
@@ -3,15 +3,14 @@
...
-    "-build": "gulp build-inferno",
-    "test": "echo \"Error: no test specified\" && exit 0"
+    "build": "gulp build-dev",
+    "watch": "gulp watch"
   },
   "author": "",
   "license": "ISC",
   "dependencies": {
-    "devextreme": "22.1.5",
-    "inferno": "^7.4.11",
```
Comment (williamvinogradov) [inline packages/jquery/package.json:13]: Do we have not ``` inferno ``` as the dependency here?

File: packages/jquery/patch/radio-button-hocs.jsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { useState, HookContainer } from '@devextreme/runtime/inferno-hooks';
```
Comment (williamvinogradov) [inline packages/jquery/patch/radio-button-hocs.jsx:1]: Is this code written by hand?

Why I'm asking about it:
E.g we change the 'handle change' method in the 'with radio group' HoC in the react component.
Should we remember about this code for jQuery and fix it here too?

Comment (MikeVitik) [reply packages/jquery/patch/radio-button-hocs.jsx:1]: Yes. I sync it with react version.

File: packages/jquery/gulp/inferno-from-react.mjs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import gulp from 'gulp';
+import gulpIf from 'gulp-if';
+import cached from 'gulp-cached';
+import rename from "gulp-rename";
+import * as ts from 'gulp-typescript';
```
Comment (IlyaKhD) [inline packages/jquery/gulp/inferno-from-react.mjs:5]: **Minor**: Isn't it possible to import `default`?
```suggestion
import ts from 'gulp-typescript';
```

**Comment by IlyaKhD** [comment]
> The npm command `npm run update:tsconfig` fails. Also, if I understood right, you touch by hand the auto-generated ts config file `packages/devextreme-core/tsconfig.package.json`
> 
> So, here are two options:
> 
> * If you need special ts configs, let's remove jQuery from the ts config generation like it was done in [this PR](https://github.com/DevExpress/DevExtreme/pull/23407/files#diff-9fd70f89265b560c377505977dd053cc8de9b67ba5b2d17ead471a5035057b72)
> * If the basic ts config for jquery is okay, let's add package.json to the `playgrounds/jQuery` and generate `packages/devextreme-core/tsconfig.package.json` by npm command.
> 
> **It'll be cool to do it separately from this PR.**

@DevExpress/devextreme-devops say we do not need `update:tsconfig` anymore. Am I get it right?
Here is a PR with deleting it: https://github.com/DevExpress/DevExtreme/pull/23416

Can these .tsconfig files be edited manually?

**Review by alexslavr — APPROVED** [review]

File: packages/jquery/gulp/inferno-from-react.mjs (hunk 6 above)
Comment (iBat) [reply packages/jquery/gulp/inferno-from-react.mjs:5]: Are we really needs gulp/babel? We already used rollup in other apps.

**Review by iBat — APPROVED** [review]

**Review by const314 — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**